src/ray/raylet/worker_pool.cc

```diff
@@ -1470,7 +1468,37 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
                    << backlog_size << " and available CPUs " << num_available_cpus
                    << " num idle workers " << state.idle.size()
                    << " num registered workers " << state.registered_workers.size();
-    PrestartDefaultCpuWorkers(task_spec.GetLanguage(), num_needed);
+    PrestartWorkersWithRuntimeEnv(task_spec, num_needed);
+  }
+}
+
+void WorkerPool::PrestartWorkersWithRuntimeEnv(const TaskSpecification &task_spec,
+                                               int64_t num_needed) {
+  RAY_LOG(DEBUG) << "PrestartWorkers " << num_needed;
+  for (int ii = 0; ii < num_needed; ++ii) {
+    GetOrCreateRuntimeEnv(
+        task_spec.SerializedRuntimeEnv(),
+        task_spec.RuntimeEnvConfig(),
+        JobID::Nil(),
+        [this, task_spec = task_spec](
+            bool successful,
+            const std::string &serialized_runtime_env_context /*unused*/,
+            const std::string &setup_error_message) {
+          if (!successful) {
+            RAY_LOG(ERROR) << "Fails to create or get runtime env "
+                           << setup_error_message;
+            return;
+          }
+          PopWorkerStatus status;
+          StartWorkerProcess(task_spec.GetLanguage(),
+                             rpc::WorkerType::WORKER,
+                             JobID::Nil(),
+                             &status,
+                             /*dynamic_options=*/{},
+                             task_spec.GetRuntimeEnvHash(),
+                             task_spec.SerializedRuntimeEnv(),
```

Comment:
this should be `serialized_runtime_env_context`

Reply:
fixed